src/agent/onefuzz-supervisor/src/scheduler.rs

```diff
@@ -131,9 +131,15 @@ impl State<SettingUp> {
     pub async fn finish(self, runner: &mut dyn ISetupRunner) -> Result<SetupDone> {
         let work_set = self.ctx.work_set;
 
-        let reboot = runner.run(&work_set).await?;
+        let output = runner.run(&work_set).await?;
 
-        let done = if reboot {
+        if let Some(output) = output {
```

Comment:
A non-success exit status is no longer an error within the `ISetupRunner`, so to preserve existing behavior, we need to scrutinize the script output here (if it exists).